crates/bevy_asset/src/io/mod.rs

```diff
@@ -269,7 +269,7 @@ pub(crate) fn get_meta_path(path: &Path) -> PathBuf {
     let mut meta_path = path.to_path_buf();
     let mut extension = path
         .extension()
-        .expect("asset paths must have extensions")
+        .unwrap_or_else(|| panic!("missing expected extension for asset path {path:?}"))
```

Comment:
This wording doesn't really make sense, did you mean this?
```suggestion
        .unwrap_or_else(|| panic!("missing extension for asset path {path:?}"))
```

Reply:
Makes sense to me but happy to change if you have a suggestion.